bftengine/src/bcstatetransfer/SourceSelector.cpp

```diff
@@ -112,13 +118,15 @@ std::string SourceSelector::preferredReplicasToString() const {
   return oss.str();
 }
 
-bool SourceSelector::shouldReplaceSource(uint64_t currTimeMilli, bool badDataFromCurrentSource) const {
+SourceReplaceType SourceSelector::shouldReplaceSource(uint64_t currTimeMilli,
+                                                      bool badDataFromCurrentSource,
+                                                      bool lastInBatch) const {
```

Comment:
Adding a default just to save modifications in existing tests is not a good reason.
I don't see a use case here for a default (unless you needed to do huge modification test).
Production calls once and explicitly always.

Reply:
I believe Anand's PR would be addressing this as I've seen such comment there.